src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -376,7 +376,7 @@ private String getPrincipalName(String username, String domainName) {
         if (slash>0) {
             principalName = username.substring(slash+1)+'@'+username.substring(0, slash)+'.'+domainName;
         } else if (username.contains("@"))
-            principalName = username+'.'+domainName;
+            principalName = username;
```

Comment:
Updated the javadoc and the case where a username contains a backslash to match the desired pattern